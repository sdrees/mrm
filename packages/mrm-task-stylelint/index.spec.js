jest.mock('fs');
jest.mock('mrm-core/src/util/log', () => ({
	added: jest.fn(),
}));
jest.mock('mrm-core/src/npm', () => ({
	install: jest.fn(),
}));

const { install } = require('mrm-core');
const { getConfigGetter } = require('mrm');
const vol = require('memfs').vol;
const task = require('./index');

const stringify = o => JSON.stringify(o, null, '  ');

const packageJson = stringify({
	name: 'unicorn',
	scripts: {
		test: 'jest',
	},
});

afterEach(() => {
	vol.reset();
	install.mockClear();
});

it('should add stylelint', () => {
	vol.fromJSON({
		'/package.json': packageJson,
	});

	task(getConfigGetter({}));

	expect(vol.toJSON()).toMatchSnapshot();
	expect(install).toBeCalledWith(['stylelint', 'stylelint-config-standard']);
});

it('should install a custom preset', () => {
	vol.fromJSON({
		'/package.json': packageJson,
	});

	task(getConfigGetter({ stylelintPreset: 'stylelint-config-pizza' }));

	expect(vol.toJSON()['/.stylelintrc']).toMatchSnapshot();
	expect(install).toBeCalledWith(['stylelint', 'stylelint-config-pizza']);
});

it('should add custom rules', () => {
	vol.fromJSON({
		'/package.json': packageJson,
	});

	task(
		getConfigGetter({
			stylelintPreset: undefined,
			stylelintRules: { 'max-empty-lines': 2 },
		})
	);

	expect(vol.toJSON()['/.stylelintrc']).toMatchSnapshot();
});

it('should use custom extension', () => {
	vol.fromJSON({
		'/package.json': packageJson,
	});

	task(getConfigGetter({ stylelintExtensions: '.sass' }));

	expect(vol.toJSON()['/package.json']).toMatchSnapshot();
});
