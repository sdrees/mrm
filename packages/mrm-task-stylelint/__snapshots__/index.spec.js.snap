// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add custom rules 1`] = `
"{
  \\"rules\\": {
    \\"max-empty-lines\\": 2
  }
}"
`;

exports[`should add stylelint 1`] = `
Object {
  "/.stylelintrc": "{
  \\"extends\\": \\"stylelint-config-standard\\"
}",
  "/package.json": "{
  \\"name\\": \\"unicorn\\",
  \\"scripts\\": {
    \\"pretest\\": \\"npm run lint:css\\",
    \\"test\\": \\"jest\\",
    \\"lint:css\\": \\"stylelint '**/*.css'\\"
  }
}",
}
`;

exports[`should install a custom preset 1`] = `
"{
  \\"extends\\": \\"stylelint-config-pizza\\"
}"
`;

exports[`should use custom extension 1`] = `
"{
  \\"name\\": \\"unicorn\\",
  \\"scripts\\": {
    \\"pretest\\": \\"npm run lint:css\\",
    \\"test\\": \\"jest\\",
    \\"lint:css\\": \\"stylelint '**/*.sass'\\"
  }
}"
`;
